methods.py

```diff
@@ -652,15 +652,15 @@ def detect_darwin_sdk_path(platform, env):
             raise
 
 
-def is_vanilla_clang(env):
+def is_apple_clang(env):
     if not using_clang(env):
         return False
     try:
```

Comment:
Let's play it safe as suggested with:
```suggestion
def is_apple_clang(env):
    if env["platform"] not in ["macos", "ios"]:
        return False
    if not using_clang(env):
        return False
    try:
```